### PR TITLE
OSSMDOC-628: make SMCP upgrade generic.

### PR DESCRIPTION
**Pull request by JStickler**
Version(s): 4.6 - 4.12

Issue: [OSSMDOC-628](https://issues.redhat.com/browse/OSSMDOC-628)
This PR makes the control plane upgrade topic generic (and moves some version specific information to the Upgrade Changes topics).  It also addresses questions asked in the JIRA.

Link to docs preview:  http://file.bos.redhat.com/jstickle/OSSMDOC-628/service_mesh/v2x/upgrading-ossm.html

Additional information:
Eng review: rcernich
QE review: pbajjuri20
Peer review:tmalove

### CODE REVIEW

**Review by rcernich — APPROVED** [review]

**Comment by JStickler** [comment]
@skondkar I think you're the upgrade reviewer for QE?

**Comment by pbajjuri20** [comment]
@JStickler : It looks good to me.

File: modules/ossm-upgrade-known-issues.adoc [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 Known issues that may affect your upgrade include:
 
-* {SMProductName} does not support the use of `EnvoyFilter` configuration except where explicitly documented. If you are using Envoy Filters, and the configuration generated by Istio has changed due to the new version of Envoy introduced by upgrading your `ServiceMeshControlPlane`, that has the potential to break any `EnvoyFilter` you may have implemented.
+* {SMProductName} does not support the use of `EnvoyFilter` configuration except where explicitly documented. This is due to tight coupling with the underlying Envoy APIs, meaning that backward compatibility cannot be maintained. If you are using Envoy Filters, and the configuration generated by Istio has changed due to the new version of Envoy introduced by upgrading your `ServiceMeshControlPlane`, that has the potential to break any `EnvoyFilter` you may have implemented.
```
Comment (tmalove) [inline modules/ossm-upgrade-known-issues.adoc:10]:
```suggestion
* {SMProductName} does not support the use of `EnvoyFilter` configuration except where explicitly documented. This is due to tight coupling with the underlying Envoy APIs, meaning that backward compatibility cannot be maintained. If you are using Envoy Filters, and the configuration generated by Istio has changed due to the latest version of Envoy introduced by upgrading your `ServiceMeshControlPlane`, that has the potential to break any `EnvoyFilter` you may have implemented.
```

Comment (tmalove) [reply modules/ossm-upgrade-known-issues.adoc:10]: This suggestion is per the entry for 'new' in the ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage

File: service_mesh/v2x/upgrading-ossm.adoc [hunk 2]
```diff
@@ -26,25 +26,26 @@ For more information about upgrading Operators, refer to the xref:../../operator
 endif::[]
 
 [id="upgrading-control-plane"]
-== Upgrading the Service Mesh control plane
+== Upgrading the control plane
 
 You must manually update the control plane for minor and major releases.
 
+While you can deploy multiple versions of the control plane in the same cluster, {SMProductName} does not support canary upgrades of the service mesh.  That is, you can have different SCMP resources with different values for `spec.version`, but they cannot be managing the same mesh.
```
Comment (tmalove) [inline service_mesh/v2x/upgrading-ossm.adoc:33]:
```suggestion
Although you can deploy many versions of the control plane in the same cluster, {SMProductName} does not support canary upgrades of the service mesh.  That is, you can have different SCMP resources with different values for `spec.version`, but they cannot be managing the same mesh.
```

Comment (tmalove) [reply service_mesh/v2x/upgrading-ossm.adoc:33]: These two changes are per the word usage list in the ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage

File: service_mesh/v2x/upgrading-ossm.adoc [hunk 3]
```diff
@@ -26,25 +26,26 @@ For more information about upgrading Operators, refer to the xref:../../operator
...
+include::modules/ossm-upgrading-smcp.adoc[leveloffset=+2]
 
-include::modules/ossm-migrating-to-20.adoc[leveloffset=+1]
+include::modules/ossm-migrating-to-20.adoc[leveloffset=+2]
 
 [id="upgrading-data-plane"]
 == Upgrading the data plane
 
-You must restart your application pods and workloads to apply updates to the Envoy proxy or changes to proxy configuration.
+Your data plane will still function after you have upgraded the control plane.  But in order to apply updates to the Envoy proxy and any changes to proxy configuration, you must restart your application pods and workloads.
```
Comment (tmalove) [inline service_mesh/v2x/upgrading-ossm.adoc:49]:
```suggestion
Your data plane will still function after you have upgraded the control plane.  But to apply updates to the Envoy proxy and any changes to the proxy configuration, you must restart your application pods and workloads.
```

Comment (tmalove) [reply service_mesh/v2x/upgrading-ossm.adoc:49]: See ISG guidelines for 'in order to'. https://www.ibm.com/docs/en/ibm-style?topic=word-usage

**Pull request merged**